src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -694,8 +697,19 @@ pQ = do
   let constructor = case quoteType of
                             SingleQuote -> B.singleQuoted
                             DoubleQuote -> B.doubleQuoted
-  withQuoteContext innerQuoteContext $
-    pInlinesInTags "q" constructor
+  let quote = withQuoteContext innerQuoteContext $ return (extractSpaces constructor lab)
+
+  -- check for cite; if cite, then a link with inner quote, otherwise a quote
+  case maybeFromAttrib "cite" tag of
+       Nothing   -> quote
+       Just url' -> do
+         let title = T.unpack $ fromAttrib "title" tag
+         -- take id from id attribute if present, otherwise name
+         let uid = fromMaybe (T.unpack $ fromAttrib "name" tag) $
+                      maybeFromAttrib "id" tag
+         let cls = words $ T.unpack $ fromAttrib "class" tag
+         url <- canonicalizeUrl url'
+         extractSpaces (B.linkWith (uid, cls, []) (escapeURI url) title) <$> quote
```

Comment:
Let's go with the span with attribute `cite` rather than the link, as discussed.